Compilation fails with MinGW-64 (gcc 4.8.1)

Firstly,

If you use mingw-64 (gcc 4.8.1) from here:
```
http://sourceforge.net/projects/mingwbuilds/
```


assimp will not compile; it complains of missing ::strlen in include/assimp/types.h


```
[  7%] Built target zlibstatic
Scanning dependencies of target assimp
[  7%] Building CXX object code/CMakeFiles/assimp.dir/Assimp.cpp.obj
C:\Users\stephen\Desktop\assimp\code\Assimp.cpp:1:0: warning: -fPIC ignored for target (all code is position indepe
ndent) [enabled by default]
 /*
 ^
In file included from C:\Users\stephen\Desktop\assimp\code\../include/assimp/Logger.hpp:47:0,
                 from C:\Users\stephen\Desktop\assimp\code\../include/assimp/DefaultLogger.hpp:46,
                 from C:\Users\stephen\Desktop\assimp\code\AssimpPCH.h:135,
                 from C:\Users\stephen\Desktop\assimp\code\Assimp.cpp:45:
C:\Users\stephen\Desktop\assimp\code\../include/assimp/types.h: In member function 'void aiString::Set(const char*)
':
C:\Users\stephen\Desktop\assimp\code\../include/assimp/types.h:285:22: error: '::strlen' has not been declared
   const size_t len = ::strlen(sz);
                      ^
C:\Users\stephen\Desktop\assimp\code\../include/assimp/types.h: In member function 'void aiString::Append(const cha
r*)':
C:\Users\stephen\Desktop\assimp\code\../include/assimp/types.h:318:22: error: '::strlen' has not been declared
   const size_t len = ::strlen(app);
```



more errors follow. This can be fixed by adding an include statement for cstring around line 48 of types.h:
```
// Some runtime headers
#include <cstring>
```

Then compilation is all good and dandy, until it tries to assemble IFCReaderGen.cpp

```
[ 53%] Building CXX object code/CMakeFiles/assimp.dir/IFCReaderGen.cpp.obj
C:\Users\stephen\Desktop\assimp\code\IFCReaderGen.cpp:1:0: warning: -fPIC ignored for target (all code is position
independent) [enabled by default]
 /*
 ^
c:/progra~1/mingw-~1/x64-48~1.1-p/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.8.1/../../../../x86_64-w64-mingw32/bi
n/as.exe: CMakeFiles\assimp.dir\IFCReaderGen.cpp.obj: too many sections (34698)
C:\Users\stephen\AppData\Local\Temp\ccvLJeF3.s: Assembler messages:
C:\Users\stephen\AppData\Local\Temp\ccvLJeF3.s: Fatal error: can't write CMakeFiles\assimp.dir\IFCReaderGen.cpp.obj
: File too big
c:/progra~1/mingw-~1/x64-48~1.1-p/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.8.1/../../../../x86_64-w64-mingw32/bi
n/as.exe: CMakeFiles\assimp.dir\IFCReaderGen.cpp.obj: too many sections (34698)
C:\Users\stephen\AppData\Local\Temp\ccvLJeF3.s: Fatal error: can't close CMakeFiles\assimp.dir\IFCReaderGen.cpp.obj
: File too big
code\CMakeFiles\assimp.dir\build.make:2673: recipe for target 'code/CMakeFiles/assimp.dir/IFCReaderGen.cpp.obj' fai
led
mingw32-make[2]: *** [code/CMakeFiles/assimp.dir/IFCReaderGen.cpp.obj] Error 1
CMakeFiles\Makefile2:158: recipe for target 'code/CMakeFiles/assimp.dir/all' failed
mingw32-make[1]: *** [code/CMakeFiles/assimp.dir/all] Error 2
Makefile:115: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```

which (I think) is because of the file being quite large (admittedly I don't know what the file is for, I'm no C++ expert. Looks auto-generated to me). I googled and came across this:
```
http://mingw-users.1079350.n2.nabble.com/File-too-big-too-many-sections-td7579168.html
```

I'd like to write Go bindings for assimp, which is why I am trying to build with mingw-64 (as it is not possible to use VS libraries with Go).

I would appreciate your time on this issue.
Stephen

Hi, I have the same problem

Win7 x64, cygwin x64 and Netbeans

 ```
/usr/lib/gcc/x86_64-pc-cygwin/4.9.3/../../../../x86_64-pc-cygwin/bin/as: CMakeFiles/assimp.dir/IFCReaderGen.cpp.o: too many sections (41590)
/cygdrive/c/Users/GBARBI~1/AppData/Local/Temp/ccmrliZV.s: Assembler messages:
/cygdrive/c/Users/GBARBI~1/AppData/Local/Temp/ccmrliZV.s: Fatal error: can't write CMakeFiles/assimp.dir/IFCReaderGen.cpp.o: File too big
/usr/lib/gcc/x86_64-pc-cygwin/4.9.3/../../../../x86_64-pc-cygwin/bin/as: CMakeFiles/assimp.dir/IFCReaderGen.cpp.o: too many sections (41590)
/cygdrive/c/Users/GBARBI~1/AppData/Local/Temp/ccmrliZV.s: Fatal error: can't close CMakeFiles/assimp.dir/IFCReaderGen.cpp.o: File too big
code/CMakeFiles/assimp.dir/build.make:2678: recipe for target 'code/CMakeFiles/assimp.dir/IFCReaderGen.cpp.o' failed
make[2]: *** [code/CMakeFiles/assimp.dir/IFCReaderGen.cpp.o] Error 1
CMakeFiles/Makefile2:172: recipe for target 'code/CMakeFiles/assimp.dir/all' failed
make[1]: *** [code/CMakeFiles/assimp.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

The similar issue with MSVS2015 x64:
qt\5.9.1_x64\qt3d\src\3rdparty\assimp\code\IFCReaderGen1.cpp : fatal error C1128: number of
sections exceeded object file format limit: compile with /bigobj
cl : Command line error D8040 : error creating or communicating with child process
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.EXE"' : return code '0x
2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\nmake.exe"' : return code
'0x2'

It blocks QT 5.9.1 compilation.

--- a/code/CMakeLists.txt
+++ b/code/CMakeLists.txt
@@ -485,9 +485,9 @@ ADD_ASSIMP_IMPORTER( IFC
 )
 if (ASSIMP_BUILD_IFC_IMPORTER)
   if (MSVC)
-    set_source_files_properties(IFCReaderGen1.cpp IFCReaderGen2.cpp PROPERTIES COMPILE_FLAGS "/bigobj")
+    set_source_files_properties(Importer/IFC/IFCReaderGen1_2x3.cpp Importer/IFC/IFCReaderGen2_2x3.cpp PROPERTIES COMPILE_FLAGS "/bigobj")
   elseif(CMAKE_COMPILER_IS_MINGW)
-    set_source_files_properties(IFCReaderGen1.cpp IFCReaderGen2.cpp PROPERTIES COMPILE_FLAGS "-O2 -Wa,-mbig-obj")
+    set_source_files_properties(Importer/IFC/IFCReaderGen1_2x3.cpp Importer/IFC/IFCReaderGen2_2x3.cpp PROPERTIES COMPILE_FLAGS "-O2 -Wa,-mbig-obj")
   endif()
 endif (ASSIMP_BUILD_IFC_IMPORTER)
 

